### ISSUE TITLE
alr toolchain --install / --install-dir now obsolete

### ISSUE THREAD

**Issue by Fabien-Chouteau** [alire-project/alire#1612]
Hi @mosteo, as part of V2 breaking changes I think we mentioned removing the install part of the toolchain command in favor of the new `alr install` command.

Is there any reason to keep that in?

**Comment by mosteo** [comment]
There are some subtle differences: 

- `alr toolchain --install` (without --install-dir) is like doing an `alr toolchain --select` but without configuring the toolchain as the default. Toolchains installed this way are available to the solver.
- `alr toolchain --install --install-dir` is more like doing `alr get`, in that it doesn't interact with Alire configuration, and they still end in a folder with funny name like `gprbuild_22.0.0_b1ab1ab1a`, and they respect the original packaging structure.
- `alr install` will create a "standard" prefix, so contents will be relocated and binaries end all inside the same `bin`. 

If we think in terms of workflow, and the idea is to have a compiler set up outside of Alire, yes, I think that between `alr install` and `alr get`, we are covered. At least `alr toolchain --install-dir` seems redundant (arguably it already was with `alr get`); it may make sense to keep `alr toolchain --install` as a way of making a toolchain available to Alire without selecting it as default.

In conclusion, I'd remove --install-dir. Maybe `--install/--uninstall` could be renamed to `--add/--del` to try to minimize confusion, and rather than talking about "installing" in the help, we could talk about "making available to Alire" or some such.

**Comment by Fabien-Chouteau** [comment]
> * `alr toolchain --install` (without --install-dir) is like doing an `alr toolchain --select` but without configuring the toolchain as the default. Toolchains installed this way are available to the solver.

The solver can still consider those releases even if they are not installed, right? So is this more like "put this in cache" command? In which case I don't think it is extremely useful.

**Comment by mosteo** [comment]
Yes, that's it. The solver tries first with the selected toolchain, then with any locally available, then with any that fits the bill, which will be downloaded if accepted.

**Comment by Fabien-Chouteau** [comment]
In that case I would also remove `--install`/`--uninstall`.
